std/algorithm/iteration.d

```diff
@@ -4257,6 +4267,11 @@ struct Permutations(Range)
 
         next(2);
     }
+
+    @property bool empty() pure nothrow @safe
```

Comment:
> Not sure what this change is doing in this PR

This was done on @andralex [request](https://github.com/D-Programming-Language/phobos/pull/4026#discussion_r54889486) to make `empty` readonly - I thought if this applies for the methods below I thought it applies for `permutation` as well.

> also @nogc is conspicuous by absence

added.